org.lflang/src/org/lflang/cli/Lfc.java

```diff
@@ -152,57 +163,53 @@ public static void main(Io io, final String... args) {
         cliMain("lfc", Lfc.class, io, args);
     }
 
-    @Override
-    protected Options getOptions() {
-        return CLIOption.getOptions();
-    }
-
     /**
      * Load the resource, validate it, and, invoke the code generator.
      */
     @Override
-    protected void runTool(CommandLine cmd, List<Path> files) {
-        Properties properties = this.filterProps(cmd, CLIOption.getPassedOptions());
-        String pathOption = CLIOption.OUTPUT_PATH.option.getOpt();
-        Path root = null;
-        if (cmd.hasOption(pathOption)) {
-            root = io.getWd().resolve(cmd.getOptionValue(pathOption)).normalize();
-            if (!Files.exists(root)) { // FIXME: Create it instead?
-                reporter.printFatalErrorAndExit("Output location '" + root + "' does not exist.");
-            }
-            if (!Files.isDirectory(root)) {
-                reporter.printFatalErrorAndExit(
-                    "Output location '" + root + "' is not a directory.");
-            }
+    public void run() {
+        List<Path> paths = getInputPaths();
+        final Path outputRoot = getOutputRoot();
+        // Hard code the props based on the options we want.
+        Properties properties = this.filterPassOnProps();
+
+        try {
+            // Invoke the generator on all input file paths.
+            invokeGenerator(paths, outputRoot, properties);
+        } catch (RuntimeException e) {
+            reporter.printFatalErrorAndExit("An unexpected error occurred:", e);
         }
+    }
 
-        for (Path path : files) {
-            if (!Files.exists(path)) {
-                reporter.printFatalErrorAndExit(path + ": No such file or directory");
-            }
-        }
+    /**
+     * Invoke the code generator on the given validated file paths.
+     */
+    protected void invokeGenerator(
```

Comment:
Why protected?

Reply:
You're right, there's no reason why it shouldn't be private. I'll fix it now.